pep/streaming_client.go

```diff
@@ -134,8 +134,15 @@ func (c *streamingClient) Validate(in, out interface{}) error {
 	}
 
 	if c.cache != nil {
-		if b, err := c.cache.Get(string(m.Body)); err == nil {
-			return fillResponse(pb.Msg{Body: b}, out)
+		var b []byte
+		if b, err = c.cache.Get(string(m.Body)); err == nil {
+			err = fillResponse(pb.Msg{Body: b}, out)
+			if err == nil {
```

Comment:
I think that the handler should be called anyway even error on unmarshalling occurs. And it should take error as well:
```go
err = fillResponse(pb.Msg{Body: b}, out)
if c.opts.onCacheHitHandler != nil {
	c.opts.onCacheHitHandler.Handle(in, out, err)
}
```
It needs to tell user that we hit cache anyway even if we can't unmarshal cached response. We even can add more complicated logic to return raw response in case of error:
```go
err = fillResponse(pb.Msg{Body: b}, out)
if c.opts.onCacheHitHandler != nil {
	if err != nil {
		c.opts.onCacheHitHandler.Handle(in, b, err)
	} else {
		c.opts.onCacheHitHandler.Handle(in, out, nil)
	}
}
```